docusaurus/docs/iOS/client/controllers/channels.md

```diff
@@ -55,9 +55,26 @@ let compoundFilter: Filter<ChannelListFilterScope> = .and([
 The `query.sort` is an array of sorting options. Sorting options are applied based on their order in the array so the first option has the highest impact while the others are used mainly as a tiebreakers. By default, the channel list is sorted by `updated_at`.
 
 ### 2. Create a controller
-
+#### With auto Channel Filtering
 The controller can be created using `ChatClient`:
 ```swift
+let controller = ChatClient.shared.channelListController(query: query)
+```
+The behaviour is controlled from the `ChatClientConfig` and more specifically the `isChannelAutomaticFilteringEnabled` flag. 
```

Comment:
I think it is better we provide a little introduction and incrementally increase the complexity, something like this:

```suggestion
### 2. Create a controller
The controller can be created using `ChatClient`:
`let controller = ChatClient.shared.channelListController(query: query)`

This is all you need to create a channel list controller, but depending on the complexity of your query, you might need to provide a custom filtering closure.

By default, the SDK will automatically handle filtering the channels as they get created. Whenever there is a web socket event that a channel has been created, the SDK will only insert it in the channel list if it matches the query. But, if the query provided contains extra data or custom filters, the SDK won't be able to automatically match the filter query, so you will need to provide a filtering closure.

#### Filtering with extra data
Here we should provide a real example of using a query + filtering closure with extra data.

#### Filtering with custom filters
Here we should provide a real example of using a query + filtering closure with a custom filter key provided.

#### Manual Filtering
Finally, here we should mention that you can disable the automatic filtering and provide a filter closure to replicate the filter logic. (We should also show a real example) 
```

Reply:
You have a very good point and I like your suggestion. Before moving forward, I need to mention that as the autofiltering is happening on the query, the data that may `arrive` in the filter closure may already not containing what the customer needs. Do you think it makes more sense to disable the auto-filtering if a filter closure has been provided?